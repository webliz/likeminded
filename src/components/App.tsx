import { useState } from 'react';
/** Components */
import Input from 'components/Layouts/FormComponents/Input/Input';
import Label from 'components/Layouts/FormComponents/Label/Label';
import Select from 'components/Layouts/FormComponents/Select/Select';
/** Styles */
import styles from './App.module.scss';
/** Types and constants */
import { initialUser } from 'components/App.types';
import { userRoles } from './App.constants';

function App() {

  const [user, setUser] = useState(initialUser);
  const [passwordFieldType, setPasswordFieldType] = useState("password");

  const handleSubmit = () => {
    console.log('Action after form submit');
  }

  return (
    <div className={styles.wrapper}>
      <div className={styles["form-wrapper"]}>
        <div className={styles.heading}>
          <div>{"Sign Up"}</div>
          <div>{"Let's create your account"}</div>
        </div>  
        <form id="likeminded-form" onClick={handleSubmit} autoComplete="off" className={styles.form}>   
          {/* Email */}
          <div className={styles["form-group"]}>
            <Label className={styles["form-label"]}>
              {"Email"}
            </Label>
              <Input
                type="text"
                name="email"
                value={user.email}
                onChange={(e:any) => {
                  setUser({...user, [e.target.name]: e.target.value });
                }}
                className={styles["form-field"]}
              />
          </div>
          {/* Password */}
          <div className={styles["form-group"]} style={{marginBottom: 50}}>
            <Label className={styles["form-label"]}>
              {"Password"}
            </Label>
            <div className={styles.password}>
              <Input
                id="password-field"
                type={passwordFieldType}
                name="password"
                value={user.password}
                onChange={(e:any) => {
                  setUser({...user, [e.target.name]: e.target.value });
                }}
                className={styles["form-field"]}
                style={{borderRadius: "8px 0 0 8px", width: "86%"}}
              />
              <div className={styles.eye} onClick={() => setPasswordFieldType(passwordFieldType === "password" ? "text" : "password")}>
                <div>
                 {passwordFieldType === "text"
                    ? <svg xmlns="http://www.w3.org/2000/svg" height={15} width={15} fill="none" viewBox="0 0 24 24" stroke="currentColor">
                        <path d="M13.875 18.825A10.05 10.05 0 0112 19c-4.478 0-8.268-2.943-9.543-7a9.97 9.97 0 011.563-3.029m5.858.908a3 3 0 114.243 4.243M9.878 9.878l4.242 4.242M9.88 9.88l-3.29-3.29m7.532 7.532l3.29 3.29M3 3l3.59 3.59m0 0A9.953 9.953 0 0112 5c4.478 0 8.268 2.943 9.543 7a10.025 10.025 0 01-4.132 5.411m0 0L21 21" />
                      </svg>
                    :
                    <svg xmlns="http://www.w3.org/2000/svg" height={15} width={15} fill="none" viewBox="0 0 24 24" stroke="currentColor">
                      <path d="M15 12a3 3 0 11-6 0 3 3 0 016 0z" />
                      <path d="M2.458 12C3.732 7.943 7.523 5 12 5c4.478 0 8.268 2.943 9.542 7-1.274 4.057-5.064 7-9.542 7-4.477 0-8.268-2.943-9.542-7z" />
                    </svg>
                  } 
                </div>
              </div>
            </div>
          </div>

          {/* Username */}
          <div className={styles["form-group"]}>
            <Label className={styles["form-label"]}>
              {"Username"}
            </Label>
            <Input
              type="text"
              name="username"
              value={user.username}
              onChange={(e:any) => {
                setUser({...user, [e.target.name]: e.target.value })
              }}
              className={styles["form-field"]}
            />
          </div>
          {/* User role */}
          <div className={styles["form-group"]}>
            <Label className={styles["form-label"]}>
              {"User role"}
            </Label>
            <Select 
              name="role"
              value={user.role}
              options={userRoles}
              onChange={(e:any) => {
                setUser({...user, [e.target.name]: parseInt(e.target.value) });
              }}
              className={styles["form-field"]}
              style={{width: "100%"}}
            />
          </div>
          <button 
            className={styles["create-action"]} 
          >
            {"Create Account"}
          </button>
        </form>
      </div>
    </div>
  );
}

export default App;
