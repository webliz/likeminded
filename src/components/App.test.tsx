import { render, screen } from '@testing-library/react';
import App from './App';
/**  Snapshots  rendering */
import renderer from 'react-test-renderer';

describe("App component", () => {

  test('renders correctly', () => {
    const component = renderer.create(<App />).toJSON();
    expect(component).toMatchSnapshot();
  });

  test('renders create account button text', () => {
    render(<App />);
    const textElement = screen.getByText(/Create Account/i);
    expect(textElement).toBeInTheDocument();
  });

});