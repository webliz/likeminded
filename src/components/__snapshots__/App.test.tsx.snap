// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="form-wrapper"
  >
    <div
      className="heading"
    >
      <div>
        Sign Up
      </div>
      <div>
        Let's create your account
      </div>
    </div>
    <form
      autoComplete="off"
      className="form"
      id="likeminded-form"
      onClick={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Email
        </label>
        <input
          className="form-field"
          data-testid="form-input"
          maxLength={30}
          name="email"
          onChange={[Function]}
          type="text"
          value=""
        />
        <div
          style={
            Object {
              "height": 4,
            }
          }
        />
      </div>
      <div
        className="form-group"
        style={
          Object {
            "marginBottom": 50,
          }
        }
      >
        <label
          className="form-label"
        >
          Password
        </label>
        <div
          className="password"
        >
          <input
            className="form-field"
            data-testid="form-input"
            id="password-field"
            maxLength={50}
            minLength={8}
            name="password"
            onChange={[Function]}
            style={
              Object {
                "borderRadius": "8px 0 0 8px",
                "width": "86%",
              }
            }
            type="password"
            value=""
          />
          <div
            className="eye"
            onClick={[Function]}
          >
            <div>
              <svg
                fill="none"
                height={15}
                stroke="currentColor"
                viewBox="0 0 24 24"
                width={15}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15 12a3 3 0 11-6 0 3 3 0 016 0z"
                />
                <path
                  d="M2.458 12C3.732 7.943 7.523 5 12 5c4.478 0 8.268 2.943 9.542 7-1.274 4.057-5.064 7-9.542 7-4.477 0-8.268-2.943-9.542-7z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "height": 8,
            }
          }
        >
          <label
            className="info-field"
          >
            Minimum 8 characters with 1 number and one uppercase letter
          </label>
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Username
        </label>
        <input
          className="form-field"
          data-testid="form-input"
          maxLength={30}
          minLength={2}
          name="username"
          onChange={[Function]}
          type="text"
          value=""
        />
        <div
          style={
            Object {
              "height": 4,
            }
          }
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          User role
        </label>
        <select
          className="form-field"
          name="role"
          onChange={[Function]}
          style={
            Object {
              "width": "100%",
            }
          }
          value={0}
        >
          <option
            value={0}
          >
            Please select a user role
          </option>
          <option
            value={1}
          >
            User
          </option>
          <option
            value={2}
          >
            Mentor
          </option>
          <option
            value={3}
          >
            Autor
          </option>
        </select>
        <div
          style={
            Object {
              "height": 4,
            }
          }
        />
      </div>
      <button
        className="create-action"
        disabled={true}
      >
        Create Account
      </button>
    </form>
  </div>
</div>
`;
